app/javascript/src/javascripts/post_flags.js

```diff
@@ -1,17 +1,176 @@
+      const selected = document.querySelector("input[name=\"post_flag[reason_name]\"]:checked");
+      return selected && parseDatasetBoolean(selected.dataset.requireExplanation) && noteField.value.trim() === "";
+    } catch (err) {
+      console.error("PostFlags.isNoteRequiredAndEmpty error", err);
+      return false;
+    }
+  }
+
+  if (form) {
+    console.debug("PostFlags: attaching submit handler to form", form);
+    form.addEventListener("submit", function (e) {
+      try {
+        updateNoteRequired();
+        if (isNoteRequiredAndEmpty()) {
+          e.preventDefault();
+          showError("Note is required for the selected reason.");
+          if (noteField) noteField.focus();
+        } else {
+          clearError();
+        }
+      } catch (err) {
+        console.error("PostFlags: submit handler error", err);
+      }
+    });
+  } else {
+    console.debug("PostFlags: form not found; submit handler not attached");
+  }
+
+  // Delegate change events from the container so switching selections always updates state
+  const flagReasonContainer = document.querySelector(".flag-reason");
+  if (flagReasonContainer) {
+    flagReasonContainer.addEventListener("change", (e) => {
+      try {
+        if (!e.target || e.target.name !== "post_flag[reason_name]") return;
+        // Remove selected class from all labels and add to the one containing the checked radio
+        flagReasonLabels.forEach(l => l.classList.remove("selected"));
+        const parentLabel = e.target.closest(".flag-reason-label");
+        if (parentLabel) parentLabel.classList.add("selected");
+        updateNoteRequired();
+      } catch (err) {
+        console.error("PostFlags: flagReasonContainer change handler error", err);
+      }
+    });
+    // Also ensure individual radios still trigger update on change
+    Array.from(radioButtons).forEach(radio => {
+      radio.addEventListener("change", updateNoteRequired);
+    });
+  } else {
+    // Fallback: attach listeners directly if container not present
+    if (radioButtons && radioButtons.length) {
+      for (let i = 0; i < radioButtons.length; i++) {
+        radioButtons[i].addEventListener("change", updateNoteRequired);
+      }
+    } else {
+      console.debug("PostFlags: no radio buttons found for reason_name");
+    }
+    if (flagReasonLabels && flagReasonLabels.length) {
+      flagReasonLabels.forEach(label => {
+        label.addEventListener("click", () => {
+          flagReasonLabels.forEach(l => l.classList.remove("selected"));
+          label.classList.add("selected");
+        });
+      });
+    }
+  }
+
+  // Run on page load in case a reason is preselected
+  updateNoteRequired();
 };
 
 export default PostFlags;
 
-$(() => {
+// Auto-initialize on standard DOM load
+if (document.readyState === "loading") {
+  document.addEventListener("DOMContentLoaded", () => PostFlags.init());
+} else {
+  // Document already loaded
   PostFlags.init();
-});
+}
```

Comment:
Jquery `$` already runs on document load. This change only makes the code longer.
```suggestion
$(() => {
  PostFlags.init();
});
```